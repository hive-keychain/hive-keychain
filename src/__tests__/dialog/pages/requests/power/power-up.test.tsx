import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import PowerUp from 'src/dialog/pages/requests/power/power-up';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestPowerUp from 'src/__tests__/utils-for-testing/data/props/dialog/request-powerUp';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
describe('power-up tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestPowerUp;
  methods.config();
  it('Must show power up dialog', async () => {
    const { asFragment } = render(<PowerUp {...props} />);
    expect(asFragment()).toMatchSnapshot('Request PowerUp with loading');
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<PowerUp {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    render(<PowerUp {...{ ...props, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    render(<PowerUp {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
