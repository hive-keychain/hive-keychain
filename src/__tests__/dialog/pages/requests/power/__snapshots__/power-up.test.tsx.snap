// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`power-up tests:
 Must show power up dialog: Request PowerUp with loading 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Power Up
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        To
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Amount
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        100 HIVE
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
